canonical/vanilla-framework
# Add reduced top navigation bar

## Done

- Moves the [reduced and secondary navigation](https://github.com/canonical/ubuntu.com/blob/main/static/sass/_pattern_navigation.scss#L413), found on u.com, over to Vanilla
- Some refactoring to align better with Vanilla defaults and remove unneeded code
- extracting sticky funtionality to it's own `is-sticky` class name

Fixes https://warthogs.atlassian.net/browse/WD-11789

## QA

Note: This is part of the work to upstream meganav functionality and may not be fully functional on its own.

Currently we focus mostly on the styling of the reduced navigation on desktop screens, making sure it doesn't break on mobile, that search is opened, that the secondary navigation below it works.

The contents of the dropdowns opened by reduced navigation are not part of this, the dropdowns funcionality on mobile is also not part of this (the "sliding" navigation).

- Open [demo](https://vanilla-framework-5107.demos.haus/docs/examples/patterns/navigation/reduced?theme=light)
- Verify that the reduced navigation (top bar), renders as expected
   - Note: "Products", "Use cases", etc - the dropdowns in reduced navigation only contain placeholder dropdowns. Only search works as it's built-in.
   - Check mobile view as well, making sure that navigations work properly and open the dropdowns, including search
   - check if secondary navigation (below reduced one) is sticky
- Review updated documentation:
  - https://vanilla-framework-5107.demos.haus/docs/patterns/navigation#reduced
  - https://vanilla-framework-5107.demos.haus/docs/patterns/navigation#sticky

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

<img width="1533" alt="image" src="https://github.com/canonical/vanilla-framework/assets/83575/e5b9b5f8-89c9-4048-ac07-27ed2388d74e">
